src/tools/android/java/com/google/devtools/build/android/r8/CompatDexBuilder.java

```diff
@@ -73,19 +75,18 @@ public void finished(DiagnosticsHandler handler) {
     }
   }
 
-  private String input;
-  private String output;
-  private int numberOfThreads = min(8, Runtime.getRuntime().availableProcessors());
-  private boolean noLocals;
-
-  public static void main(String[] args)
-      throws IOException, InterruptedException, ExecutionException {
-    new CompatDexBuilder().run(args);
+  public static void main(String[] args) throws Exception {
+    CompatDexBuilder compatDexBuilder = new CompatDexBuilder();
+    compatDexBuilder.processRequest(args);
   }
 
   @SuppressWarnings("JdkObsolete")
-  private void run(String[] args) throws IOException, InterruptedException, ExecutionException {
+  private void processRequest(String[] args) throws Exception {
```

Comment:
Why not add OptionsParsingException instead of using the much wider Exception?

Reply:
I'll revert this part

Reply:
Updated!